ledger/participant-integration-api/src/main/scala/platform/store/backend/common/CommonStorageBackendFactory.scala

```diff
@@ -8,20 +8,12 @@ import com.daml.platform.store.cache.LedgerEndCache
 
 trait CommonStorageBackendFactory extends StorageBackendFactory {
 
-  override def createPackageStorageBackend(ledgerEndCache: LedgerEndCache): PackageStorageBackend =
-    new PackageStorageBackendTemplate(ledgerEndCache)
-
   override val createParameterStorageBackend: ParameterStorageBackend =
     ParameterStorageBackendTemplate
 
   override val createMeteringParameterStorageBackend: MeteringParameterStorageBackend =
     MeteringParameterStorageBackendTemplate
 
-  override def createConfigurationStorageBackend(
-      ledgerEndCache: LedgerEndCache
-  ): ConfigurationStorageBackend =
-    new ConfigurationStorageBackendTemplate(ledgerEndCache)
```

Comment:
What about having an abstract `def queryStrategy: QueryStrategy` so that these two methods don't have to be moved out to three separate backends?